File: 345-reverse-vowels-of-a-string/345-reverse-vowels-of-a-string.go
func reverseVowels(s string) string {
	result := make([]byte, len(s))
	vowelsMap := [10]string{"a", "A", "e", "E", "i", "I", "o", "O", "u", "U"}

	i, j := 0, len(s)-1
	for i <= j {
		if !checkVowels(vowelsMap, string(s[i])) {
			result[i] = s[i]
			i++
			continue
		}

		if !checkVowels(vowelsMap, string(s[j])) {
			result[j] = s[j]
			j--
			continue
		}

		result[i] = s[j]
		result[j] = s[i]
		i++
		j--
	}

	return string(result)
}

func checkVowels(vowels [10]string, s string) bool {
	for _, v := range vowels {
		if v == s {
			return true
		}
	}
	return false
}

File: 345-reverse-vowels-of-a-string/345-reverse-vowels-of-a-string.js
var reverseVowels = function (s = "") {
  let result = [];
  let i = 0, j = s.length - 1;

  while (i <= j) {
    if (!checkVowels(s[i])) {
      result[i] = s[i];
      i++;
      continue;
    }

    if (!checkVowels(s[j])) {
      result[j] = s[j];
      j--;
      continue;
    }

    result[i] = s[j];
    result[j] = s[i];
    i++;
    j--;
  }

  return result.join('');
};

var checkVowels = (c) => {
  return ['a', 'e', 'i', 'o', 'u'].includes(c?.toLowerCase())
}